containers/src/Kokkos_ErrorReporter.hpp

```diff
@@ -57,71 +96,62 @@ class ErrorReporter {
       typename Kokkos::View<
           report_type *, typename DeviceType::execution_space>::host_mirror_type
           &reports_out);
+#endif
 
-  void clear();
+  bool full() const { return (num_report_attempts() >= capacity()); }
 
-  void resize(const size_t new_size);
+  void clear() const {
+    int num_reports = 0;
+    Kokkos::deep_copy(m_numReportsAttempted, num_reports);
+  }
 
-  bool full() { return (getNumReportAttempts() >= getCapacity()); }
+  void resize(const size_t new_size) {
+    Kokkos::resize(m_reports, new_size);
+    Kokkos::resize(m_reporters, new_size);
+    typename DeviceType::execution_space().fence(
+        "Kokkos::Experimental::ErrorReporter::resize: fence after resizing");
+  }
 
   KOKKOS_INLINE_FUNCTION
   bool add_report(int reporter_id, report_type report) const {
```

Comment:
If you are going for container-like interface consider calling it push_back, insert, emplace or whatnot
Since it may fail may be prefixed by try_*

Reply:
Probably the clearest semantics would be having `try_add_report` and `add_report` where the letter errors out on full.